Pass `_allow_empty_param_list` into func opt ctor

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63164 Remove req to call step() in training loop
* **#63163 Pass `_allow_empty_param_list` into func opt ctor**
* #63162 Simplify data structures, add uniform approximation, fix mem leak

Differential Revision: [D30284615](https://our.internmc.facebook.com/intern/diff/D30284615)

**Overview:**
This passes `_allow_empty_param_list` into the functional optimizer constructor, taking advantage of the recently-landed PR: https://github.com/pytorch/pytorch/pull/62522. This helps avoid erroring in cases where a rank is assigned none of the parameters to update.

Commit:
Pass `_allow_empty_param_list` into func opt ctor

[ghstack-poisoned]

andwgu commented:
@andwgu has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30284615).

mrshenli reviewed (approved):
LGTM

Commit:
Update on "Pass `_allow_empty_param_list` into func opt ctor"


Differential Revision: [D30284615](https://our.internmc.facebook.com/intern/diff/D30284615)

**Overview:**
This passes `_allow_empty_param_list` into the functional optimizer constructor, taking advantage of the recently-landed PR: https://github.com/pytorch/pytorch/pull/62522. This helps avoid erroring in cases where a rank is assigned none of the parameters to update.

[ghstack-poisoned]

andwgu commented:
@andwgu has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30284615).
